src/pip/_internal/cli/cmdoptions.py

```diff
@@ -130,6 +130,17 @@ def check_dist_restriction(options, check_target=False):
             )
 
 
+def _path_option_check(option, opt, value):
+    # type: (Option, str, str) -> str
+    return os.path.expanduser(value)
+
+
+class _PathOption(Option):
```

Comment:
```suggestion
class PipOption(Option):
```

As it will most probably evolve to support other types than `path`.